### DISCUSSION TITLE
Fun with Capcha

### DISCUSSION THREAD

**Discussion by tony-in-nz** [primefaces/community#1923] [PrimeFaces] [answered]
I spent a long time trying to get Capcha working on a new project.

I found that the PF doco refers to a legacy way of having the public/privaye key pair.

I found that going into the google configuration screens the wrong way and trying to set up a new key pair just ends up being incredibly painful as you go around and around and never get see the secret (hours wasted).

I wanted to post this URL so people can have less pain:

https://cloud.google.com/security/products/recaptcha?hl=en
[Get Started]
Add label (site name only)
Add domain (site)
leads to site and secret key

I am not sure how long this will be around

**Comment by melloware** [comment, marked as answer]
So our PF docs need updating?  Can you provide an update?

PrimeFaces 15.0.0 adds support for HCaptcha as well.
